crates/bevy_xr/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+pub mod interaction;
+pub mod presentation;
+
+use bevy_app::{AppBuilder, Plugin, StartupStage};
+use bevy_ecs::prelude::{IntoExclusiveSystem, World};
+use interaction::{
+    GenericControllerPairButtons, GenericControllerVibration, HandAction, HandType, Motion, Pose,
+    Position, TrackingReferenceMode, XR_HAND_JOINT_COUNT,
+};
+
+pub struct XrDuration(i64);
+
+impl XrDuration {
+    pub fn from_nanos(nanos: i64) -> Self {
```

Comment:
Is there a reason we can't use `std::time::Duration` instead of `i64`?

Reply:
And then you could just do `impl Deref for XrDuration` where `Target = std::time::Duration`.

Reply:
The problem with `std::time::Duration` is that it cannot be negative and a naked `i64` has no unit of measure attached which might be important if working at higher level. `impl Deref for XrDuration` cannot be implemented because it would fail if `XrDuration` and this goes against the `Deref` impl convention. Instead the user should use `XrDuration::to_nanos()` + `Duration::from_nanos()`.

Reply:
Well what I was suggesting (sorry for not clarifying) was to define `XrTime` as
```
pub struct XrTime(std::time::Duration);
```

And then have the `Deref` built on top of that

Reply:
Also using an `i64` is a little weird because _negative time_ isn't really a thing.
Hence why `std::time::Duration::from_nanos` uses an `u64`

Reply:
This makes more sense. I'll try it.

Reply:
@NathanSWard I think negative `Duration` is an omission from std. Very frequently negative durations could arise from subtraction between `Instant`s. Dealing with an always positive `Duration` becomes tricky (and this is shown in std). Anyway, to be honest, much discussion here shoud not be needed since `XrTime` and `XrDuration` have a very small use in this API. I'm also fine to drop both `XrTime` and `XrDuration` and use std versions.

Reply:
Yea, I'm just always wary of introducing new `duration`/`instant` types. 
And about _negative durations_ it is something that `std` does not handle for example even if you subtract two instants that would result in a negative duration, the implementation panics. 

If we really want to support them, we should consider using the [chrono](https://docs.rs/chrono/0.4.19/chrono/index.html) library and not rolling our own.